buildSrc/src/main/groovy/com/google/firebase/gradle/UploadMeasurementsTask.groovy

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.firebase.gradle
+
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+import org.gradle.api.DefaultTask
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Takes the size information created by {@link GenerateMeasurementTask} and uploads it to the
+ * database using the uploader tool.
+ *
+ * <p>The uploader tool is fetched from the Internet using a URL. This URL, and the path to the
```

Comment:
What's to prevent someone from spamming this service using the tool ?

Reply:
The uploader tool is hosted on Google Cloud Storage. If we encounter a problem with spam we may need to rethink that or restrict file access.